GameObject.hpp

```diff
@@ -128,7 +151,14 @@ namespace spic {
              */
             template<class T>
             std::shared_ptr<Component> GetComponent() const {
-                // ... implementation here
+                if(std::is_base_of<Component, T>::value){
+                    for(Component component: components){
+                        if(typeid(component) == typeid(T)){
+                            return std::make_shared<Component>(component);
+                        }
+                    }
+                }
```

Comment:
```cpp
if(!(std::is_base_of<Component, T>::value))
	return nullptr;

for(Component component: components){
	if(typeid(component) == typeid(T)){
		return std::make_shared<Component>(component);
	}
}
```

Notice how I do an early return instead of nesting the for loop inside the if statement.